presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1061,12 +1061,16 @@ else if (HadoopFileStatus.isDirectory(fileStatus)) {
     @Override
     public void createView(ConnectorSession session, SchemaTableName viewName, String viewData, boolean replace)
     {
-        if (replace) {
-            try {
-                dropView(session, viewName);
-            }
-            catch (ViewNotFoundException ignored) {
+        Optional<Table> existing = metastore.getTable(viewName.getSchemaName(), viewName.getTableName());
+        if (existing.isPresent()) {
+            Table table = existing.get();
+            if (!replace || !HiveUtil.isPrestoView(table)) {
+                throw new ViewAlreadyExistsException(viewName);
             }
+
+            table.setViewOriginalText(encodeViewData(viewData));
+            metastore.alterTable(viewName.getSchemaName(), viewName.getTableName(), table);
```

Comment:
I don't think this is correct.  Instead of modifying the existing `Table` object, build a new one using the code below and then call alter with the new table.